### ISSUE TITLE
Support LIMIT with prefetch and MySQL

### ISSUE THREAD

**Issue by nomme** [coleifer/peewee#2631]
Hi

The premise is simple. It would be pretty sweet if peewee's `prefetch` supports `LIMIT` when using MySQL. I want to add support for it by introducing a switch on `prefetch` that subqueries using `INNER JOIN` instead of `IN`.

Background:
In a project using peewee and MySQL we often encounter the following limitation when using `prefetch`.
```
This version of MySQL doesn't yet support 'LIMIT & IN/ALL/ANY/SOME subquery'
```

This can be seen reported multiple times across the internet, which you also address all those times. [There are ways around it](https://github.com/coleifer/peewee/pull/381#issuecomment-47823210) but native support would feel much better, and would also require one less query and avoid giant `IN` lists with large limit values.

I'll submit a PR in a moment to show what I have in mind.

**Comment by coleifer** [comment]
Merged your PR here: af3f34f10555a22003726dca90fb2d577617656a

**Comment by nomme** [comment]
Awesome, thank you!
